src/cargo/ops/fix/mod.rs

```diff
@@ -588,7 +588,15 @@ fn check_resolver_change<'gctx>(
             feature_opts,
         )?;
 
-        let diffs = v2_features.compare_legacy(&ws_resolve.resolved_features);
+        if ws_resolve.specs_and_features.len() != 1 {
+            bail!(r#"Cannot fix edition when using `feature-unification = "projects"`."#);
```

Comment:
> cargo fix - from what I understand, the location is reached only with --edition. I think it should be ok to disallow using cargo fix --edition when someone already uses this feature.

This should just be limited to projects with `edition = "2018"` which should be fine